internal/namingschema/op_client_server.go

```diff
@@ -37,13 +37,32 @@ type serverInboundOp struct {
 	system string
 }
 
+type SimpleSchema func() string
+
+func (s SimpleSchema) Name() string { return s() }
+
+type Namer interface {
+	Name() string
+}
+
 // NewServerInboundOp creates a new naming schema for server inbound operations.
-func NewServerInboundOp(system string, opts ...Option) *Schema {
+func NewServerInboundOp(system string, opts ...Option) Namer {
```

Comment:
This is more or less a hack. It's not consistent with the rest of the namingschema package.

We should refactor the whole package to do something similar to this.